airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/DatabaseConfigPersistence.java

```diff
@@ -227,17 +232,141 @@ private int insertConfigRecord(DSLContext ctx, OffsetDateTime timestamp, String
     return 1;
   }
 
-  @Override
-  public Map<String, Stream<JsonNode>> dumpConfigs() throws IOException {
-    LOGGER.info("Exporting all configs...");
+  /**
+   * @return the number of updated records.
+   */
+  @VisibleForTesting
+  int updateConfigRecord(DSLContext ctx, OffsetDateTime timestamp, String configType, JsonNode configJson, String configId) {
+    LOGGER.info("Updating {} record {}", configType, configId);
 
-    Map<String, Result<Record>> results = database.query(ctx -> ctx.select(asterisk())
+    return ctx.update(AIRBYTE_CONFIGS)
+        .set(AIRBYTE_CONFIGS.CONFIG_BLOB, JSONB.valueOf(Jsons.serialize(configJson)))
+        .set(AIRBYTE_CONFIGS.UPDATED_AT, timestamp)
+        .where(AIRBYTE_CONFIGS.CONFIG_TYPE.eq(configType), AIRBYTE_CONFIGS.CONFIG_ID.eq(configId))
+        .execute();
+  }
+
+  @VisibleForTesting
+  void copyConfigsFromSeed(DSLContext ctx, ConfigPersistence seedConfigPersistence) throws SQLException {
+    LOGGER.info("Loading data to config database...");
+
+    Map<String, Stream<JsonNode>> seedConfigs;
+    try {
+      seedConfigs = seedConfigPersistence.dumpConfigs();
+    } catch (IOException e) {
+      throw new SQLException(e);
+    }
+
+    OffsetDateTime timestamp = OffsetDateTime.now();
+    int insertionCount = seedConfigs.entrySet().stream().map(entry -> {
+      String configType = entry.getKey();
+      return entry.getValue().map(configJson -> {
+        String idFieldName = ConfigSchemaMigrationSupport.CONFIG_SCHEMA_ID_FIELD_NAMES.get(configType);
+        return insertConfigRecord(ctx, timestamp, configType, configJson, idFieldName);
+      }).reduce(0, Integer::sum);
+    }).reduce(0, Integer::sum);
+
+    LOGGER.info("Config database data loading completed with {} records", insertionCount);
+  }
+
+  @VisibleForTesting
+  void updateConfigsFromSeed(DSLContext ctx, ConfigPersistence seedConfigPersistence) throws SQLException {
+    LOGGER.info("Config database has been initialized; updating connector definitions from the seed if necessary...");
+
+    try {
+      Set<String> usedConnectorRepos = getUsedConnectorRepos(ctx);
```

Comment:
```suggestion
      Set<String> usedConnectorRepos = getExistingConnectorRepos(ctx);
```

Reply:
`used` made me think we no longer want to use this

Reply:
I think `existing` can be a bit misleading here. All these connector repositories "exist" in the database. Some of them are used in standard syncs, while some are not. I can replace it with `in use`.